RecoTracker/LSTCore/src/alpaka/Quintuplet.h

```diff
@@ -1699,8 +1690,28 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE::lst {
                                   const float ptCut) const {
       ALPAKA_ASSERT_ACC((alpaka::getWorkDiv<alpaka::Grid, alpaka::Blocks>(acc)[1] == 1) &&
                         (alpaka::getWorkDiv<alpaka::Grid, alpaka::Blocks>(acc)[2] == 1));
+
+      int& matchCount = alpaka::declareSharedVar<int, __COUNTER__>(acc);
+
+      const auto threadIdx = alpaka::getIdx<alpaka::Block, alpaka::Threads>(acc);
+      const auto blockDim = alpaka::getWorkDiv<alpaka::Block, alpaka::Threads>(acc);
+
+      const int threadIdX = threadIdx[2];
+      const int threadIdY = threadIdx[1];
+      const int blockSizeX = blockDim[2];
+      const int blockSizeY = blockDim[1];
+      const int blockSizeZ = blockDim[0];
+      const int blockSize = blockSizeX * blockSizeY * blockSizeZ;
+      const int flatThreadIdxXY = threadIdY * blockSizeX + threadIdX;
+      const int flatThreadExtent = blockSize;  // total threads per block
```

Comment:
are you sure you should include the Z block size ?
the Z thread index is _not_ included.

Reply:
The Z block size is 1 anyways, so include It or not is the same

Reply:
Including it only in one of the variables is conceptually wrong and confusing.

Please remove it from the definition of `flatThreadExtent`.

Reply:
ok, I will remove it. Thanks